includes/utilities.inc

```diff
@@ -22,6 +22,38 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+/**
+ * Autocomplete callback; return all objects that can be an association.
+ */
+function islandora_ead_get_potential_associations($search_value) {
+
+  $label_field = variable_get('islandora_solr_object_label_field', 'fgs_label_s');
+
+  // Make the query.
+  $qp = new IslandoraSolrQueryProcessor();
+  $query = islandora_solr_lesser_escape("$search_value*");
+  $qp->buildQuery($query);
+  $qp->solrParams['fl'] = "$label_field,PID";
+  $qp->executeQuery(FALSE);
+
+  // Helper function to construct labels.
+  $construct_label = function ($data) use ($label_field) {
+    $label = isset($data['solr_doc'][$label_field]) ? $data['solr_doc'][$label_field] : '';
+    return t('@label (@pid)', array(
+      '@label' => $label,
+      '@pid' => $data['solr_doc']['PID'],
+    ));
+  };
+
+  $results = array();
+  foreach ($qp->islandoraSolrResult['response']['objects'] as $result) {
+    $results[$result['PID']] = $construct_label($result);
+  }
+  ksort($results);
```

Comment:
Not sure this `ksort` is necessary here? Could take advantage of Solr's sort field such that we don't need to sort in memory in PHP.